### PR TITLE
feat(transforms): Add docs for string join transform

### PR DESCRIPTION
**Pull request by MisterMX**

### CODE REVIEW

File: content/master/concepts/patch-and-transform.md [hunk 1]
```diff
@@ -1654,6 +1655,25 @@ patches:
           trim: `-north-1'
 ```
 
+#### Join
```
Comment (plumbis) [inline content/master/concepts/patch-and-transform.md:1658]: Can you move this section to be before line 1601, the heading "Regular expression type" to make the type order alphabetical?

File: content/master/concepts/patch-and-transform.md [hunk 2]
```diff
@@ -1654,6 +1655,25 @@ patches:
...
+
+The {{<hover label="typeJoin" line="8">}}type: Join{{</hover>}} joins all
+values in the input array into a string using the given separator.
```
Comment (plumbis) [inline content/master/concepts/patch-and-transform.md:1661]:
```suggestion
values in the input array into a string using the given separator.  
```
I'd suggest adding a line break to put the line "This transform only works..." on a new line.

File: content/master/concepts/patch-and-transform.md [hunk 3]
```diff
@@ -1546,22 +1546,23 @@ patches:
...
-String transforms support the following 
+String transforms support the following
 {{<hover label="string" line="7">}}types{{</hover>}}
 
 * [Convert](#string-convert)
 * [Format](#string-format)
 * [Regexp](#regular-expression-type)
 * [TrimPrefix](#trim-prefix)
 * [TrimSuffix](#trim-suffix)
+* [Join](#join)
```
Comment (plumbis) [inline content/master/concepts/patch-and-transform.md:1557]: Can you move this to be the third item in the list so it's alphabetical?

**Review by plumbis — CHANGES_REQUESTED** [review]
The new content looks good, however looks like your IDE may have chopped off all ending spaces which changes formatting in markdown. Can you restore the removed spaces?

Do we need to commit this to 1.14 as well or is this only in master today?

**Comment by plumbis** [comment]
_ping _ @MisterMX

File: content/master/concepts/patch-and-transform.md (hunk 3 above)
Comment (MisterMX) [reply content/master/concepts/patch-and-transform.md]: Moved it up.

File: content/master/concepts/patch-and-transform.md (hunk 2 above)
Comment (MisterMX) [reply content/master/concepts/patch-and-transform.md]: Added.

File: content/master/concepts/patch-and-transform.md (hunk 1 above)
Comment (MisterMX) [reply content/master/concepts/patch-and-transform.md]: Done.

**Comment by MisterMX** [comment]
@plumbis I applied your requested changes. Can you take a second look?

**Review by plumbis — APPROVED** [review]

**Pull request merged**